uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -1,44 +1,59 @@
 define([
   'leave-absences/shared/modules/models-instances',
-  'leave-absences/shared/models/leave-status-id-model',
+  'common/services/api/option-group',
   'common/models/instances/instance'
 ], function (instances) {
   'use strict';
 
   instances.factory('LeaveRequestInstance', [
     'ModelInstance',
     'LeaveRequestAPI',
-    'LeaveStatusID',
-    function (ModelInstance, LeaveRequestAPI, LeaveStatusID) {
+    'api.optionGroup',
+    function (ModelInstance, LeaveRequestAPI, OptionGroup) {
+
+      /**
+       * This method is used to get ID of an option value
+       *
+       * @param {string} name - name of the option value
+       * @return {Promise} Resolved with {Object} Specific leave request
+       */
+      function getOptionIDByName(name) {
+        return OptionGroup.valuesOf('hrleaveandabsences_leave_request_status')
+          .then(function (data) {
+            return data.find(function (statusObj) {
+              return statusObj.name === name;
+            })
+          })
+      }
+
       return ModelInstance.extend({
 
         /**
          * This method is used to cancel a leave request
-         *
-         * @return {Promise}
          */
         cancel: function () {
-          var leaveRequest = this;
-          return LeaveStatusID.getOptionIDByName("cancelled")
+          return getOptionIDByName('cancelled')
             .then(function (cancelledStatusId) {
-              return leaveRequest.update({
-                'status_id': cancelledStatusId
-              })
-            })
+              return this.update({
+                'status_id': cancelledStatusId.value
+              });
+            }.bind(this))
             .then(function (data) {
-              return data.values[0];
-            });
+              if (data.is_error === 1) {
+                return data;
```

Comment:
Why do you return the data if there has been an error? Throw an exception containing the error message, instead